### PR TITLE
Radial layout: center on root option

### PR DESCRIPTION
**Pull request by Eddykasp**
This option allows a radial to be automatically centered on its root node. This is achieved by increasing the overall size of the layout and shifting all nodes accordingly.

### CODE REVIEW

File: plugins/org.eclipse.elk.alg.radial/src/org/eclipse/elk/alg/radial/intermediate/CalculateGraphSize.java [hunk 1]
```diff
@@ -48,21 +50,58 @@ public void process(final ElkNode graph, final IElkProgressMonitor progressMonit
...
         KVector offset = new KVector(minXPos - padding.getLeft(), minYPos - padding.getTop());
+        
+        
+        double width = maxXPos - minXPos + padding.getHorizontal();
+        double height = maxYPos - minYPos + padding.getVertical();
+        
+        if (graph.getProperty(RadialOptions.CENTER_ON_ROOT)) {
+            ElkNode root = graph.getProperty(InternalProperties.ROOT_NODE);
+            ElkMargin rootMargins = root.getProperty(CoreOptions.MARGINS);
+            double rootX = root.getX() + root.getWidth()/2 + (rootMargins.left + rootMargins.right)/2 - offset.x;
```
Comment (soerendomroes) [inline plugins/org.eclipse.elk.alg.radial/src/org/eclipse/elk/alg/radial/intermediate/CalculateGraphSize.java:61]: Maybe add a small explanation for this formula

File: plugins/org.eclipse.elk.alg.radial/src/org/eclipse/elk/alg/radial/intermediate/CalculateGraphSize.java [hunk 2]
```diff
@@ -48,21 +50,58 @@ public void process(final ElkNode graph, final IElkProgressMonitor progressMonit
...
+        }
 
         // process the nodes
         for (ElkNode node : graph.getChildren()) {
             // set the node position
             node.setX(node.getX() - offset.x);
             node.setY(node.getY() - offset.y);
         }
-
+
```
Comment (soerendomroes) [inline plugins/org.eclipse.elk.alg.radial/src/org/eclipse/elk/alg/radial/intermediate/CalculateGraphSize.java:97]: Additional whitespace

**Review by soerendomroes — CHANGES_REQUESTED** [review]
Have you considered adding a test for this?

**Comment by Eddykasp** [comment]
> Have you considered adding a test for this?

I thought you might ask that. I'll add that and address the other things as well.

File: test/org.eclipse.elk.alg.radial.test/.project [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>org.eclipse.elk.alg.disco.test</name>
```
Comment (soerendomroes) [inline test/org.eclipse.elk.alg.radial.test/.project:3]: Why is this project called disco.test?

File: test/org.eclipse.elk.alg.radial.test/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,14 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: ELK Layout DisCo Tests
```
Comment (soerendomroes) [inline test/org.eclipse.elk.alg.radial.test/META-INF/MANIFEST.MF:3]: Name

File: test/org.eclipse.elk.alg.radial.test/build.properties [hunk 5]
```diff
@@ -0,0 +1,13 @@
+###############################################################################
+# Copyright (c) 2019 Kiel University and others.
```
Comment (soerendomroes) [inline test/org.eclipse.elk.alg.radial.test/build.properties:2]: Wrong year

File: test/org.eclipse.elk.alg.radial.test/pom.xml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright (c) 2017 Kiel University and others.
```
Comment (soerendomroes) [inline test/org.eclipse.elk.alg.radial.test/pom.xml:3]: Wrong year

**Review by soerendomroes — CHANGES_REQUESTED** [review]

**Review by soerendomroes — APPROVED** [review]

**Pull request merged**